packaging/language/pip.py

```diff
@@ -413,15 +433,14 @@ def main():
             changed = False
             if name:
                 for pkg in name:
-                    is_present = _is_present(pkg, version, out.split())
+                    is_present = _is_present(pkg, version, out.split('\n'))
                     if (state == 'present' and not is_present) or (state == 'absent' and is_present):
                         changed = True
                         break
             module.exit_json(changed=changed, cmd=freeze_cmd, stdout=out, stderr=err)
 
         if requirements or has_vcs:
-            freeze_cmd = '%s freeze' % pip
-            out_freeze_before = module.run_command(freeze_cmd, cwd=chdir)[1]
+            _, _, out_freeze_before, _ = _get_packages(module, pip, chdir)
```

Comment:
Need to add some error checking here.  Before, wasn't robust but it probably did the job.  It checked that the 3-tuple of output was the same before and after.  Now, we're only saving stdout so we're not capturing whether there was an error condition or the package just has no installable packages to report.  Need to be explicit about checking the rc value now (and probably reporting stdout+stderr in the fail_json msg)